WordPress/Classes/Services/CommentService.m

```diff
@@ -111,46 +112,47 @@ - (Comment *)restoreReplyForComment:(Comment *)comment
                     failure:(void (^)(NSError *error))failure
 {
     NSManagedObjectID *blogID = blog.objectID;
     if (![[self class] startSyncingCommentsForBlog:blogID]){
         // We assume success because a sync is already running and it will change the comments
         if (success) {
-            success();
+            success(YES);
         }
         return;
     }
     
     id<CommentServiceRemote> remote = [self remoteForBlog:blog];
-    [remote getCommentsWithSuccess:^(NSArray *comments) {
-                             [self.managedObjectContext performBlock:^{
-                                 Blog *blogInContext = (Blog *)[self.managedObjectContext existingObjectWithID:blogID error:nil];
-                                 if (blogInContext) {
-                                     [self mergeComments:comments
-                                                 forBlog:blog
-                                           purgeExisting:YES
-                                       completionHandler:^{
-                                           [[self class] stopSyncingCommentsForBlog:blogID];
-                                           
-                                           blogInContext.lastCommentsSync = [NSDate date];
-                                           [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
-                                           
-                                           if (success) {
-                                               success();
-                                           }
-                                       }];
-                                 }
-                             }];
-                         } failure:^(NSError *error) {
-                             [[self class] stopSyncingCommentsForBlog:blogID];
-                             if (failure) {
-                                 [self.managedObjectContext performBlock:^{
-                                     failure(error);
-                                 }];
-                             }
-                         }];
+    [remote getCommentsWithMaximumCount:WPNumberOfCommentsToSync success:^(NSArray *comments) {
+         [self.managedObjectContext performBlock:^{
+             Blog *blogInContext = (Blog *)[self.managedObjectContext existingObjectWithID:blogID error:nil];
+             if (blogInContext) {
+                 [self mergeComments:comments
+                             forBlog:blog
+                       purgeExisting:YES
+                   completionHandler:^{
+                       [[self class] stopSyncingCommentsForBlog:blogID];
+                       
+                       blogInContext.lastCommentsSync = [NSDate date];
+                       [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
+                       
+                       if (success) {
+                           BOOL hasMore = comments.count >= WPNumberOfCommentsToSync;
```

Comment:
Might just be me missing something, but I couldn't quite follow why this `hasMore` calculation is different from the one on [line 201](https://github.com/wordpress-mobile/WordPress-iOS/pull/4705/files#diff-14c270a91165f087b12949de7fb679edR201). Could you perhaps stick a comment in to clarify?